src/EditorFeatures/Core/Implementation/ExtractMethod/AbstractExtractMethodCommandHandler.cs

```diff
@@ -72,27 +72,28 @@ public VSCommanding.CommandState GetCommandState(ExtractMethodCommandArgs args)
 
         public bool ExecuteCommand(ExtractMethodCommandArgs args, CommandExecutionContext context)
         {
-            var document = args.SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
-            if (document == null)
+            using (context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Applying_Extract_Method_refactoring))
             {
-                return false;
-            }
+                var document = args.SubjectBuffer.CurrentSnapshot.GetFullyLoaddedOpenDocumentInCurrentContextWithChangesAsync(
+                    context.OperationContext).WaitAndGetResult(context.OperationContext.UserCancellationToken);
+                if (document == null)
+                {
+                    return false;
+                }
 
-            var supportsFeatureService = document.Project.Solution.Workspace.Services.GetService<IDocumentSupportsFeatureService>();
-            if (!supportsFeatureService.SupportsRefactorings(document))
-            {
-                return false;
-            }
+                var supportsFeatureService = document.Project.Solution.Workspace.Services.GetService<IDocumentSupportsFeatureService>();
+                if (!supportsFeatureService.SupportsRefactorings(document))
+                {
+                    return false;
+                }
```

Comment:
this should be front loaded.  that's why i had a GetFullyLoadedDocumentThatSupportsRefactoring.  You could also have something like `GetFullyLoadedDocument(bool mustSupportRefactoring)`.

Reply:
this was not done.